app/models/post.rb

```diff
@@ -1,12 +1,22 @@
 class Post < ApplicationRecord
   # Attributes
   attribute :title, :string
-  attribute :tetxt, :text
+  attribute :text, :text
   attribute :comments_counter, :integer
   attribute :likes_counter, :integer
 
   # Associations
-  belongs_to :author, class_name: 'User'
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  # Methods
+  after_create :update_user_posts_counter
```

Comment:
- [ ] Great job in using one of Rails Active Callback `after_create`. However, after_create is not triggered when an existing record is updated. In this case, I suggest using `after_save` which is called both after the `create` and `update` operations.
- In Ruby on Rails, callbacks are often defined as `private methods `within the model class to encapsulate the logic and make them accessible only within the class. This helps in keeping the code organized and prevents the callback methods from being accidentally called from outside the class.
Therefore the update_user_posts_counter should be made private. Your model should look like this:
```ruby
  after_save :update_user_posts_counter

  private

  def update_user_posts_counter
    author.increment!(:posts_counter)
  end
```